code/modules/wiremod/components/math/arctan2.dm

```diff
@@ -0,0 +1,26 @@
+/**
+ * # Arctangent 2 function
+ * A variant of arctan. When given a deltaX and deltaY, returns the angle. I will blow you out of the sky
+ */
+/obj/item/circuit_component/arctan2
+	display_name = "Arctangent 2 Component"
+	desc = "A two parameter arctan2 component, for calculating any angle you want."
+	category = "Math"
+
+	/// The input port for the x-offset
+	var/datum/port/input/input_port_x
+	/// The input port for the y-offset
+	var/datum/port/input/input_port_y
+
+	/// The result from the output
+	var/datum/port/output/output
+
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
+
+/obj/item/circuit_component/arctan2/populate_ports()
+	input_port_x = add_input_port("Input", PORT_TYPE_NUMBER)
+	input_port_y = add_input_port("Input", PORT_TYPE_NUMBER)
```

Comment:
Should these not say "X" and "Y"?

Reply:
Do they not or am I tripping?

Reply:
I'm referring to the names--this is just "Input" and "Input". Shouldn't they be "X" and "Y"?

Reply:
Oh, maybe?